GCTC-NTGC/gc-digital-talent
# Add azure  Application Insights instrumentation to IAP pages

Resolves #3219.



### How to Test Local
1. Check out this branch
2. Add `"InstrumentationKey=8f4f1fd8-8831-4a3b-84a0-20245c86e418;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/"` as the value for `APPLICATIONINSIGHTS_CONNECTION_STRING` in .env 
3. Open the network tab in your browser
4. Navigate to '/en/indigenous-it-apprentice'
5. Search for **track** in the network tab and verify the headers and payload

### To Do
- [x] Investigate better place to store connection string (secrets?) 
- [x] Deploy in the dev environment so we can see the metrics ( created a jira for team access to this AppInsights resource in Azure https://jira.tbs-sct.gc.ca/browse/AZ-13533 ) 
- [ ] Could we split the instrumentation as a sep wrapper from page container in IAP? So we can reuse for talencloud folder as well. ( for team/code reviewers to comment on. )  

### Azure Metrics screenshot 
![image](https://user-images.githubusercontent.com/89482910/185218205-173d72c9-c49d-444b-bddb-e3ea43e6f2d0.png)


**Deployment Notes**

### Dev & UAT 

Add the below environment variable
 APPLICATIONINSIGHTS_CONNECTION_STRING=""  

( empty string as we don't want to collect metrics from these environments ) 


### Prod 

Prod needs to be given the actual connection string from Azure AppInsights. You can get it directly from @brindasasi OR 
By this time deployment of this story all devs would have those access in Azure. 
Here is how you can collect the connection string 
<img width="1018" alt="migrate-from-instrumentation-keys-to-connection-strings (1)" src="https://user-images.githubusercontent.com/89482910/185683834-852c3d23-fb9b-4755-8799-0bc014f2ae4b.png">
https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net

 APPLICATIONINSIGHTS_CONNECTION_STRING= "[original connection string from azure] "
